### DISCUSSION TITLE
TX Timeout hardcoded

### DISCUSSION THREAD

**Discussion by ffraporti** [Lora-net/LoRaMac-node#1580] [Q&A]
https://github.com/Lora-net/LoRaMac-node/blob/6384af8f4c4a57016c83f3d34430f0f68855e695/src/mac/region/RegionAU915.c#L560C94-L560C99

Should the TXTimeout be hard coded to 4000?

In Brazil, the standard RX delay is 5000ms, if you send retry for confirmed messages, it should be after the rx window closes. It might be better to get this timeout based on the region parameters.

**Comment by EduardoPfeifer** [comment]
> https://github.com/Lora-net/LoRaMac-node/blob/6384af8f4c4a57016c83f3d34430f0f68855e695/src/mac/region/RegionAU915.c#L560C94-L560C99
> 
> Should the TXTimeout be hard coded to 4000?
> 
> In Brazil, the standard RX delay is 5000ms, if you send retry for confirmed messages, it should be after the rx window closes. It might be better to get this timeout based on the region parameters.

Hi @ffraporti 

The RX Delay only opens after a transmission. I think the timeout is a safe time for the radio, as it is possible to send an uplink with almost 3 seconds of air time, so 4 seconds of timeout I think is not a problem.
